include/dos_system.h

```diff
@@ -45,17 +45,6 @@ constexpr auto CurrentDirectory = ".";
 constexpr auto ParentDirectory  = "..";
 constexpr auto DosSeparator     = '\\';
 
-// obsolete - TODO: remove
-enum FatAttributeFlagsValues : uint8_t { // 7-bit
-	DOS_ATTR_READ_ONLY = 0b000'0001,
-	DOS_ATTR_HIDDEN    = 0b000'0010,
-	DOS_ATTR_SYSTEM    = 0b000'0100,
-	DOS_ATTR_VOLUME    = 0b000'1000,
-	DOS_ATTR_DIRECTORY = 0b001'0000,
-	DOS_ATTR_ARCHIVE   = 0b010'0000,
-	DOS_ATTR_DEVICE    = 0b100'0000,
-};
-
 union FatAttributeFlags {
 	uint8_t _data = 0;
```

Comment:
(just a minor suggestion, if you think it's better)

One of the nice things with enums is that the definition is a "spaceless" logical construct, without being real objects.

So replacing this `enum` with the global objects is kind of trade off - given (I think) we would still /like/ to use them as enums, but unfortunately C++ doesn't allow non-scalar enums.

Ideally we'd have:

```c++
enum class FatAttributes : FatAttributeFlags {
    ReadOnly = { 0b000'0001 },
    Hidden   = { 0b000'0010 },
    System   = { 0b000'0100 },
    Archive  = { 0b000'1000 },
    ...
};
```

One alternative (that get's us a bit closer) is to make a scoped enum inside the union:

```c++
union FatAttributeFlags {
	enum : uint8_t {
		ReadOnly  = bit::literals::b0,
		Hidden    = bit::literals::b1,
		System    = bit::literals::b2,
		Volume    = bit::literals::b3,
		Directory = bit::literals::b4,
		Archive   = bit::literals::b5,
		Device    = bit::literals::b6,
		NotVolume = bit::mask_flip_all(Volume),
	};
    ...
};
```

This eliminates the global objects and it's strongly scope  just like a class-enum.

The tradeoff is we still need to use scalars (uint8_t) in enum, which matches the `_data` type.

This allows for a bit more natural comparisons when dealing with the uint8_t and uint16_t `"attr"` and `"attrib"` variables, and we can still assign and pass them into functions that take `FatAttributeFlags` objects.

Reply:
@FeralChild64 , I've pushed a branch off your branch to show it in action, if you'd like to cherry-pick:

Here's the change for you to review: 1570dee6d

The branch `kc/fc-attr-improvements-scoped-enum-1` has two commits, if you want to git cherry-pick them:

`git cherry-pick ac60fb05d` // white-space only
`git cherry-pick 1570dee6d` // scoped-enum

Reply:
Very nice @kcgen, I think that's an improvement over the const approach, and it's nicely namescaped. Good creative use of namespace nesting! 🏅

Reply:
OK, makes sense. Cherry-picket the commit.